### PR TITLE
adds telemetry for analytics

### PR DESCRIPTION
**Pull request by re-Tick**
#### Describe the changes you've made
added a telemetry service calls to a localhost analytics server to collect anonymous user data.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

File: pkg/platform/telemetry/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	"net/http"
+	"time"
+
+	"go.keploy.io/server/pkg/models"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.uber.org/zap"
+)
+
+type Telemetry struct {
+	TelemetryDB DB
```
Comment (slayerjain) [inline pkg/platform/telemetry/telemetry.go:14]: the variable could also just be named `db`

File: pkg/platform/telemetry/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+			EventType: eventType,
+			CreatedAt: time.Now().Unix(),
+		}
+		// only 1 or no meta is passed in output array parameter
+		if len(output) == 1 {
+			event.Meta = output[0]
+		}
+		sr := ac.TelemetryDB.Find()
+		if sr == nil {
+			ac.logger.Error("")
```
Comment (slayerjain) [inline pkg/platform/telemetry/telemetry.go:108]: why are you logging an empty string?

File: pkg/platform/telemetry/telemetry.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	if ac.Enabled {
+		event := models.Event{
+			EventType: eventType,
+			CreatedAt: time.Now().Unix(),
+		}
+		// only 1 or no meta is passed in output array parameter
+		if len(output) == 1 {
+			event.Meta = output[0]
+		}
+		sr := ac.TelemetryDB.Find()
```
Comment (slayerjain) [inline pkg/platform/telemetry/telemetry.go:106]: you should just save this id while initializing the telemetry service. This way you can avoid hitting the database for every telemetry call

File: pkg/service/regression/regression.go [hunk 4]
```diff
@@ -35,7 +37,7 @@ func New(tdb models.TestCaseDB, rdb run.DB, log *zap.Logger, EnableDeDup bool) *
 
 type Regression struct {
 	tdb models.TestCaseDB
-
+	adb telemetry.Service
```
Comment (slayerjain) [inline pkg/service/regression/regression.go:40]: adb doesn't sound right

**Review by slayerjain — CHANGES_REQUESTED** [review]
awesome work @re-Tick. Once we sort out these minor issues we should be good to merge. 

This seems to be well thought out and written! 👍

File: pkg/platform/telemetry/service.go [hunk 5]
```diff
@@ -11,7 +11,7 @@ import (
 type DB interface {
 	Count() (int64, error)
 	Insert(id string) (*mongo.InsertOneResult, error)
```
Comment (slayerjain) [inline pkg/platform/telemetry/service.go:13]: the telemetry package shouldn't depend on the mongo package

**Review by slayerjain — CHANGES_REQUESTED** [review]
Looks fine mostly.

**Review by Sarthak160 — APPROVED** [review]

**Comment by nehagup** [comment]
@slayerjain is this still moving?

**Comment by re-Tick** [comment]
Added testcases for telemetry. Please review.

File: pkg/platform/telemetry/telemetry.go (hunk 1 above)
Comment (re-Tick) [reply pkg/platform/telemetry/telemetry.go]: Done

File: pkg/platform/telemetry/telemetry.go (hunk 2 above)
Comment (re-Tick) [reply pkg/platform/telemetry/telemetry.go]: Sorry forgot to remove it.

File: pkg/service/regression/regression.go (hunk 4 above)
Comment (re-Tick) [reply pkg/service/regression/regression.go]: renamed it to tele. is this fine

**Review by slayerjain — APPROVED** [review]

**Pull request merged**